### ISSUE TITLE
Flash camera video not working on Chrome or Firefox

### ISSUE THREAD

**Issue by edualcw** [ruffle-rs/ruffle#5046]
I use a camera with a Flash website. Now that Flash is gone, I tried the ruffle emulation but the screen just stays black.

**Comment by parkerlreed** [comment]
Camera API access is part of AS3 which is not yet fleshed out in ruffle

**Comment by midgleyc** [comment]
Camera access is also part of AS2, see #244

If you post the SWF / website, that would be helpful for testing.

**Comment by edualcw** [comment]
Thanks for this info. The camera is on a local network and only one user can be configured to access it. If I open a port and give the credentials, everybody could change the password and I couldn't access it anymore. But meanwhile I installed a Mac server with Sierra and downloaded an old version of Flash so that I can still see the camera through this server until it will work with the ruffle emulator.

**Comment by parkerlreed** [comment]
Oh so this isn't for webcam access? As mentioned, do you have an example of the website/software? (Can the viewer be downloaded by anyone and just inspected). 

@midgleyc Thanks for the clarification. I had a feeling it was available earlier but wasn't seeing it right off hand.

**Comment by edualcw** [comment]
No, it is just a personal surveillance camera with limited access.

**Comment by nicolas2k** [comment]
Hi!

It's same for me with surveillance cameras Corsee (CH)...
I don't found alternative ...

**Comment by edualcw** [comment]
For now, I solved the issue by uploading an old version of Flash on a Mac with High Sierra or similar OS.

**Comment by nicolas2k** [comment]
I've tried to use abdobe-flash plugins from under Chrome or Firefox, nothing has been worked for AS3, however the demos work. I used under Ubuntu 20.04 patched or Windows Build 21H1 but I don't have a Mac. I didn't found some old package that you suggest for this platform like above.

**Comment by midgleyc** [comment]
You might be able to use [Basilisk Portable](https://archive.org/details/basilisk-portable-with-flash) for Flash.

**Comment by edualcw** [comment]
I installed Flash for Mac OsX Safari-Firefox V. 32.0.0.330, disallowed updates and it works perfectly, but only on older OsX versions, not on Catalina nor Big Sur.

**Comment by edualcw** [comment]
> You might be able to use [Basilisk Portable](https://archive.org/details/basilisk-portable-with-flash) for Flash.

There seems to be no Mac version available.

**Comment by edualcw** [comment]
And Firefox V.72 works with this version of Flash on Catalina...but you have to remove automatic updates immediately.

**Comment by c933103** [comment]
http://www.ditiezu.com/home.php?mod=spacecp&ac=avatar

This Chinese forum, which require Flash for uploading images as forum avatar, does not work using ruffle due to the lack of camera support.

**Comment by Lord-McSweeney** [comment]
Closing as no SWF file or Ruffle logs was included. Feel free to reopen the issue if you can upload either.
